### ISSUE TITLE
Does RESTier have a future, and when?

### ISSUE THREAD

**Issue by rendmath** [OData/RESTier#540]
We couldn't help but notice that Microsoft seems to have stopped all investment in RESTier for multiple months now. The library is still very far away from feature parity with WCF Data Services. The last commit is now 2 months old and no new major feature has been implemented for the last 6 months.

We understand that Microsoft doesn't have unlimited resources. But neither do we, and we can't afford to invest our limited resources in a seemingly discontinued project. If we need to revert to WCF Data Services, we would like to know it as soon as possible.

We also understand that the technologies RESTier depends upon are themselves in a state of flux (complete rewrite of Entity Framework and ASP.NET with EF7 and .NET Core), and Microsoft may not want to invest more resources in frameworks that are already being replaced. Nevertheless, we need to know what to expect.

Is Microsoft still committed to pursue the development of RESTier and bring it to feature parity with WCF Data Services?
Will Microsoft continue to develop RESTier for both ASP.NET 4 and ASP.NET Core, and both Entity Framework 6 and Entity Framework 7 ?

**Comment by fileman** [comment]
... or is another LightSwitch? (many features in common)

**Comment by peder** [comment]
Wow, I hadn't realized the extent of the problem. Looks like [Vincent He](https://github.com/chinadragon0515?tab=overview&from=2016-11-04) has stopped contributing to GitHub completely? He seemed like an army of one.

We're in a similar spot trying to determine which libraries to use, since the entire .NET ecosystem seems chaotic at the moment. It's interesting that you mention parity with WCF Data Services, since server-side WCF also appears to have been abandoned in favor of REST.

We could use honest roadmaps from Microsoft to help us plan. There are a ton of libraries that are still heavily used, but there doesn't appear to be a plan to support them in .NET Core nor is there a suitable alternative available.

**Comment by peder** [comment]
I looked through all the contributions of the OData team... everything slowed down in August. Was the team disbanded?

**Comment by robertmclaws** [comment]
Guys, it's important to note that the same team is responsible for WebAPI, OData, and Restier. I e-mailed some people on the team to see what they're working on. As soon as I hear anything I'll report back.

**Comment by mikepizzo** [comment]
The stewardship of the OData core libraries, including RESTier, is transitioning to a new team at Microsoft. 

We had hoped to make the transition as smooth as possible, and the development team that has owned OData for the past 3 years has been exceptional in supporting the transition, but obviously the changes are being felt by the community while the new team gets resourced and up to speed.

OData remains an important technology to Microsoft (in fact, the organizational changes were to move it to a group where it was more aligned with long term strategic vision) and the transition is expected to be completed in January at which time we should have more to share.  Until then, thank you for your patience and please bear with us.

Thanks for your ongoing support and interest,

 Michael Pizzo
 Principal Architect, OData
 Microsoft
 Editor, OData Specification
 OASIS

**Comment by peder** [comment]
Thanks for the update, Michael. I think a lot of the .NET community is left to try and divine the direction of .NET in face of the disruptive .NET Core and open-source movement, so it's great to hear the direction.

**Comment by mikepizzo** [comment]
Support for .NET Core is on our radar, but will be a fair amount of work in the OData WebAPI layer used by RESTier.

**Comment by MichelZ** [comment]
It's mid-January now... any updates on this?

**Comment by vincedan** [comment]
need ms update the news.. please.

**Comment by andyminnitt** [comment]
Good day Mr. Pizzo,

Any update on the future of RESTier, it seems like such a powerful project, we have stared to use it for a fairly substantial Web Application which will integrate into our company's ERP system.

Please tell me that RESTier is alive and well :-)

**Comment by rendmath** [comment]
@mikepizzo I see there has been some recent movement in ODataLib and OData Web API. Can we hope for a roadmap of any kind regarding RESTier ? At this point, even bad news would be preferable to no news at all.

**Comment by WinInsider** [comment]
@mikepizzo I do hope to see RESTier make a comeback on .NET Standard...

**Comment by jwobbeoya** [comment]
@mikepizzo Add me to the list of people hoping to get an updated status on this project.

**Comment by fileman** [comment]
@mikepizzo seem that RESTier and Lightswitch have not only features in common, now we are [here](https://blogs.msdn.microsoft.com/lightswitch/2014/12/05/where-we-are-and-the-road-ahead/) when is planned to reach [next step](https://blogs.msdn.microsoft.com/lightswitch/2016/10/14/lightswitch-update/)?

> fileman commented on Nov 25 2016 
... or is another LightSwitch? (many features in common)

**Comment by fileman** [comment]
Signs of life?
https://www.nuget.org/packages/WinInsider.Microsoft.Restier.Core/

**Comment by WinInsider** [comment]
@fileman this is a unofficial Restier packages recompile for .NET Standard... while MS gets up to speed

**Comment by khuzemakanore** [comment]
Hello

Any update please?

**Comment by robertmclaws** [comment]
@WinInsider I'd like to use your Restier.Core package. How do you use it with Restier?

Thanks!

**Comment by WinInsider** [comment]
I will write a doc in a day or so; however, while Restire.Core and EF.Core Provider is ~99% same, functionality of ODataPublisher is largely diffrent.  Namely, is not based WebAPI, but rather on OWIN... It allows to execute LINQ queries from client via HTTP GET.

**Comment by schmidtaa** [comment]
Colleagues at MSFT, if you have updates especially around roadmap, please share. This is helpful for architecturing solutions based on latest Microsoft technology stack, otherwise we have to remove risks and removing those libraries from consideration.

**Comment by robertmclaws** [comment]
Prefacing all of this by saying, once again, that I do not speak for the team. I don't know what their status is on roadmap, but I think they are hard at work on OData .NET Core issues. Not sure if you guys noticed, but they did a MAJOR refactor of the core libraries that has been working its way up through the system. Check NuGet, they are on Version 7.1.1 now, and 7.2 builds are in their Nightly feed.

I have a PR (#560) that I'm trying to usher through their process now. I'm going to see if I can get the team to ship all of the recent changes (including my reference updates and some fixes) as an updated beta. At the moment, they are trying to help me track down a bug that is preventing EntitySet filters from working on one of my services. Would like to get that in there so no one else hits it.

HTH!

**Comment by fileman** [comment]
@robertmclaws I think that taking the time to write "we're working on it" even though they are working hard to migrate to .net core, they might find it and I think it would be enough to most of those who commented on this.
Just to not do as with LightSwitch, canceled after 2 years of silence ... OData Team blog's last post is 9th August 2016.
Anyway good job to the team, we look forward to use updated versions as soon as possible ... meanwhile I continue to work with my customized 1.0.0 beta, hoping for a smooth migration to RTM!

**Comment by robertmclaws** [comment]
Alright guys, I've pushed [my fork of Restier](https://github.com/robertmclaws/RESTier) onto NuGet. You can grab it here: https://www.nuget.org/packages/advancedrei.restier/ The server side includes a bunch of fixes and some small refactors from [this pull request](https://github.com/OData/RESTier/pull/560).

THIS IS A USE AT YOUR OWN RISK RELEASE. Back up your stuff. If you have an issue, add it to my fork and I'll see what I can do to get it fixed.

It is important to note that Microsoft does not have a compatible client library for the server-side binaries we're using (7.1.1). There is a new client in the 7.2 nightlies that I'm trying to test against. Older client generators should still work if they are using the older assemblies. I'm working on getting the T4 templates updated, but Visual Studio 2017.2's T4 debugging is busted, so I'll have to get to it later this week.

Now, when Microsoft does actually get around to shipping something, I'll update my NuGet packages to point to Microsoft's packages, so you shouldn't miss a beat.

To the OData team: I thought long and hard about whether or not to do this. In the end, my love for this platform and my responsibility to the community as an MVP is too great to wait, when my quality bar to ship is much lower. I'd still like to help get a more official 1.0 Beta 2 out the door.

Hit me up if any of you have any questions. HTH!

**Comment by robertmclaws** [comment]
FYI, I pushed an update today because one of my NuGet packages pointed to the wrong OData.Core package. If you decided to try out my fork, please update.

**Comment by nmoreaud** [comment]
Any news ? Any reaction from Microsoft ?

**Comment by robertmclaws** [comment]
Microsoft is busy working on getting the lower parts of the OData stack on .NET Standard. They have the core libraries in place, and now they're working on Web API and the Client generators. I imagine they will get to Restier after that.

In the meantime I am moving forward with my fork. I've also built a testing framework to make it easier to build reliable Restier APIs. At the moment, I'm working on getting Swashbuckle.OData working, to connect Restier with DocFX, as well as unit tests against the Swagger output. More details on that soon.

I've asked Microsoft for permission to move my work into a branch of this repo instead of my personal fork.. we'll see what they say.

If I hear anything else, I'll let you all know.

**Comment by mikepizzo** [comment]
Just a quick note to give a sign of life -- we haven't forgotten you, and hope soon to be in a position to be much more responsive. Thanks for sticking with us this far!

**Comment by adiuvo** [comment]
I love the idea, but I cannot even begin to justify using it in the current state. I tried, it was not pretty :-)

**Comment by tudort-iquest** [comment]
Indeed, it's a shame that there is no clear roadmap and no consistent feedback from Microsoft team..
After WCF Data Services were abandoned, this looked like a possible alternative, but it doesn't seem the case.

**Comment by bkwdesign** [comment]
I'm still loving OData v4.. my current documentation I'm providing to my team instructs the developer to first fallback to V.S. 2015 which supports the **OData v4 Web API Scaffolding** VSIX to allow my team members to quickly generate odata api controllers and MVC controllers+Views...  

Furthermore, this VSIX works with the Nuget: `Microsfot.AspNet.Odata`  **v5.10.0** (and not 6.x.+)

Once we've successfully generated our controllers and razor code in VS2015, we move move it all over into our production WebAPI and MVC projects that are in VS 2017, (using the latest `Microsfot.AspNet.Odata`) which we then deploy to Azure. 

The only issue w/scaffolding, is that our MVC controllers are meant to do data-access though the API, so, we have to refactor the generated code a bit to ensure they make api calls instead of Entity Framework calls..  but the scaffolding is SO NICE for a starting point.

**Comment by philcontrolf1** [comment]
@mikepizzo "we [...] hope soon to be in a position to be much more responsive."

Six months on, there hasn't been much communication on this project. I think people would appreciate any update even if it's just "sorry, we changed our priorities".

**Comment by robertmclaws** [comment]
@philcontrolf1 You guys should be paying attention to the main OData repos. They've been working to get OData working on both ASP.NET and ASP.NET Core with shared libraries. It's been a massive re-architecture, and they are nearly done. Once they are, they will turn their attention back to this project to get it shipped.

**Comment by tudort-iquest** [comment]
@robertmclaws - WCF Data Services was practically abandoned at the beginning of 2014 (https://blogs.msdn.microsoft.com/odatateam/2014/03/27/discussion-future-direction-of-wcf-data-services/ )
Restier was announced at the end of the same year (2014): http://www.odata.org/blog/restier-a-turn-key-framework-to-build-restful-service/
Now (january 2018), more than 3 years later, we still don't have a final version and a replacement for WCF Data Services. :(

**Comment by robertmclaws** [comment]
I understand, and I feel your pain. That's why I had to fork the project and ship my own version on NuGet (look up AdvancedREI.Restier). it's also why I shipped my own docs (restier.readthedocs.io).

@tudort-iquest The team has gone through a lot of turmoil over the last 3 years. Microsoft de-prioritized it and moved it to a junior team in China. Then they re-prioritized it and moved it bac to the USk. Then they screwed over everyone with this fork of .NET called .NET Core, forcing the OData team to re-architect everything in order to placate the screaming OData users that couldn't move to .NET Core.

The wait is almost over. Follow the WebAPI repo if you want to keep your eye on the progress.

**Comment by tudort-iquest** [comment]
I can understand that there are objective reasons for the delay, the problem is the lack of communication and feedback on this matter.

**Comment by fileman** [comment]
asp.net core webapi odata is in Beta2 ... Restier no progress or not visible to us ... we have to start migration on our own?

**Comment by cilerler** [comment]
I just wanted to inform you all that latest nightly build of **Microsoft.AspNet.OData** `v7.0.0-Nightly201805241245` allows us to use RESTier with **Microsoft.Extensions.DependencyInjection** `v2.0.0`.  

Nightly builds are available on https://www.myget.org/F/webapinetcore/api/v3/index.json

You may download the latest RESTier with the necessary changes from [here](https://github.com/cilerler/RESTier) _(tests are broken and I added EF7 into build script)_ however if you want to have original repository [here](https://github.com/xuzhg/RESTier/tree/UpdateToWebAPIOData7x) it is.

### Steps to follow
1. Open the `Restier.EF7` solution _(regardless of your usage of EF7)_
2. Change the configuration to `Release` 
3. Right click on `src` folder and click `Rebuild`
4. Make sure you are not getting any error message on **Output** window.
5. Open the solution directory from **Command Prompt**
5. Run the command `BuildNuGets.cmd Release`
6. Your new nuget files will be in **bin\nuget\Release** folder
7. Update your project with those _(add local folder as NUGET source)_
8. Replace all `System.Web.OData` with `Microsoft.AspNet.OData` in your solution

Thank you very much @xuzhg moving the issue from P4 to P1 

![bitmoji](https://render.bitstrips.com/v2/cpanel/02719625-0b2f-4b1f-81fa-797787740e60-1d989a62-ae8a-48a8-b04f-0f575358153a-v1.png?transparent=1&palette=1&width=246)

**Comment by xuzhg** [comment]
@cilerler you're welcome.

**Comment by tudort-iquest** [comment]
@cilerler - that's great news - maybe an announcement at https://blogs.msdn.microsoft.com/odatateam/ would help, otherwise many companies won't start using it until they will see a final 1.0 version announced there.

**Comment by cwoodruff** [comment]
Thanks for the news about RESTier! I am digging into it this week.

**Comment by Tiberriver256** [comment]
Thanks for the news. This is an awesome project.

**Comment by Tiberriver256** [comment]
@cilerler any update for us? When could we expect development resources to return to RESTier?

**Comment by cilerler** [comment]
@xuzhg do you think you may gather inside info on the question above? Any update would be appreciated, including **it will be not in the 2019 road-map**.  Thanks in advance...

**Comment by Tiberriver256** [comment]
@xuzhg - Know you're busy. Just a gentle ping on this.

**Comment by MichelZ** [comment]
After this item has been open for 2 years, I think for all intents and purposed this project is DEAD. Even if they release new stuff, the communication policy for this project is so terrible, that I'm definitely NOT going to take a bet on this one for the future. Goodbye RESTier.   REST in piece.

**Comment by robertmclaws** [comment]
@MichelZ @Tiberriver256 @cilerler etc. I totally understand where you're coming from. Don't give up hope yet. The project is in stasis right now (not dead), but I'm personally very invested in seeing 1.0 shipped, as I've built an entire business around this framework and am supporting it for the State of Florida, among other entities. The right discussions are happening right now about this project, and that's all I can say for now. Hopefully there will be more concrete direction by the end of the year.

**Comment by mikepizzo** [comment]
Sorry for the quiet; sometimes things move more slowly than we'd like. We do appreciate the enthusiasm from this community. As @robertmclaws says, we've been investing in the lower level libraries to enable support in RESTier for things like .NET Core and many new features from OData 4.01. We now have the foundation we need in place, and are actively working on a plan to unblock progress on this fundamental project.

**Comment by tudort-iquest** [comment]
Let's be serious.. :) Restier was started more than 4 years ago, as a replacement of the abandoned WCF Data Services and is still not ready - if it is not ready for production already, it will never be.

Unfortunately the truth is that Microsoft practically abandoned the OData idea when they abandoned Silverlight 6 years ago, and WCF Data Services (Astoria project) was created in 2007 for Silverlight. :(

**Comment by Tiberriver256** [comment]
@robertmclaws / @mikepizzo - Thank for the responses and it's great to hear. Would it be possible to get someone (community or MSFT internal) that would be able to monitor PRs on this project in the mean time?

There are a few old PRs that were submitted that may help move the project forward in some small way.

**Comment by robertmclaws** [comment]
@Tiberriver256 I know I have at least one PR that needs to be integrated as well. I've been sitting on a bunch of work in other branches, including significant improvements to the docs, and guidance on architecture and code generation. It all depends on what Microsoft wants to do.

@tudort-iquest You're welcome to your opinion, but that's far from the truth. OData was never abandoned, it suffered from a re-org that moved the project over to China for a couple years. I have no inside info, but I think management forgot how many MSFT teams use OData, and it languished for a little while. But they get it now.. the problem is the whole architecture had to be rewritten. The new team back at Redmond has made massive progress in a gigantic ecosystem, but they still have a lot of work ahead of them.

Instead of being Debbie-Downers, we need to show Microsoft there is still support for this and the other OData projects, so they will continue to invest in Making OData Great Again.

**Comment by tudort-iquest** [comment]
@robertmclaws  - indeed, it's just a subjective opinion - most people just go to https://blogs.msdn.microsoft.com/odatateam/ and see no posts between 2016 and this year..
The documentation from https://github.com/OData/odata.github.io saw no updates in the last 2-3 years

**Comment by mikepizzo** [comment]
In the interest of open communication, though we haven’t closed on the details, I wanted to provide a (long overdue) update to this community.

Having long followed this discussion, which is a reflection of the ongoing deep frustration within an active and enthusiastic community, and personally feeling that frustration, I have been searching for a solution to unblock RESTier from moving forward.

To be completely candid; Microsoft has been in an ongoing quandary. While we continue to make significant investment in the lower level OData libraries used across Microsoft products and services, and while those investments benefit and are leveraged by RESTier, we have been unsuccessful in internally justifying the necessary level of investment in RESTier, based on current usage of that framework within Microsoft (most of whom are today using the lower level WebAPI OData and OData .NET libraries). Still, we have been unwilling to let go of the project as we see tremendous potential in the stack and believe that many teams inside Microsoft, as well as externally, would benefit from the framework if it was available. 

Unfortunately, failing both to provide a statement of direction, and to apply resources to manage the contributions of the community, has effectively blocked, and in many cases alienated, those members that have been the most supportive and engaged, and with whom we would ideally be partnering to build the best possible framework.

So, in an effort to unblock the community and move RESTier forward to GA, we are looking at how we can open the project up more fully to the community. The details and logistics are still being worked out, but long time contributors @robertmclaws and @cwoodruff  have agreed to step up to help manage contributions, releases, etc.  I expect that we will grow the inner team of administrators over time, but it is important to me that we identify a few key folks committed to move the framework forward, and I am fully confident that it is in good hands with Robert and Chris.

Microsoft will continue to be involved (in fact, this discussion has rekindled an interest within Microsoft to dedicate resources to the project) but I believe it’s important that Microsoft not be a blocker, and instead that we fully empower a team of external contributors to be equal administrators in the project.

I am personally excited to see the project move forward, and to take advantage of the significant ongoing work that has taken place in the underlying libraries as well as integrate much of the ongoing work from external contributors.  And, I want to thank the community that has stuck with the project, and Robert and Chris in particular for being willing to step into an administrative role.

As I say, the details are still being worked out, but I felt that the community deserved a long overdue update on our thinking and our intentions.

**Comment by cilerler** [comment]
This is an amazing news @mikepizzo 
I'm glad that we are going back to the business...  
**Long live RESTier!**

**Comment by Tiberriver256** [comment]
Excellent, I am sure you will see enough community contributions to keep this rolling!

**Comment by fileman** [comment]
great news, because @robertmclaws probably has more experience with RESTier than MS "USA" team

**Comment by madansr7** [comment]
I want to thank each one of you out there for using and contributing to OData. A big shout out to @robertmclaws and @cwoodruff for agreeing to step in and help us out. 
We acknowledge there are gaps in documentation, and we haven't paid a lot of attention to the blogs either. 

The project transition from China along with the new committed work, attributed to the low prioritization of our docs and blogs. Given the time and resources, we prioritized .NET Core and other long-awaited features on top which affected the docs contribution. But we hear you. Good features need good docs!

It’s tough to top the announcement @mikepizzo just made, but I can tell you we have plans to revise our documentation and blogs. Starting next year, you will see a lot of movement on at end. We are going through a revision of the docs, so please stay tuned for more information.

If you have ideas or things that you would like to have included in the documentation, please ping me. We are all ears about incorporating changes that will make life easy for OData developers.   

Again, the OData team is glad to have a community like you. 

On behalf of the entire team, I would like to wish you all happy holidays and happy new year!

**Comment by robertmclaws** [comment]
@madansr7 Thanks for the post, and happy holidays to you too! Am definitely going to want to chat with you about documentation. I'm not sure what strategy you folks are coming up with right now, but I do know the RESTier documentation has been historically sub-par, and I intend to fix that for the 1.0 release. My tentative plan was to migrate my docs at https://restier.readthedocs.io to DocFX and retire what is at http://odata.github.io/RESTier, so it can be replaced with something much more comprehensive. Those plans are still in their very early stages though, and haven't been cleard with MSFT yet.

I'm still getting the team discussion group set up, but let's circle back before the end of the year and chat some more. :) Thanks so much!

**Comment by robertmclaws** [comment]
Hey Everyone! Just wanted to let you know that I'm hard at work on Restier this weekend, and am banging through issues on the [new project board](https://github.com/OData/RESTier/projects/1) for the V1 release.

I know there are a bunch of other issues in the project, but the first goal is to modernize the codebase so we can get a working build on the very latest OData code, and then see what issues are left. I've actually made a crapload of progress on that, and should be able to check something in soon.

As soon as I get permissions on the OData organization, I'll have an Azure Pipeline set up and cranking out CI builds for the Beta 2 release.

Onward!

**Comment by cilerler** [comment]
> My tentative plan was to migrate my docs at https://restier.readthedocs.io to DocFX and retire what is at http://odata.github.io/RESTier, so it can be replaced with something much more comprehensive. Those plans are still in their very early stages though, and haven't been cleared with MSFT yet.

@robertmclaws do you think we may go with Wiki part of the GitHub until product becomes more mature. I bet Microsoft has plans to make GitHub's Wiki part compatible with DocFX, 

@superyyrrzz can you give us any inside info about that subject?

**Comment by robertmclaws** [comment]
@cilerler We can do whatever the community wants, but I will say that DocFX powers docs.microsoft.com and is pretty mature. GitHub runs independent of Microsoft and whatever their plans are for the Wiki, I doubt that's going to happen in the next 2-3 weeks. When I said I wanted to get another build out the door soon, I wasn't kidding. ;) So we may have to make due with something else for 1.0 and then do something better for 2.0.

**Comment by cilerler** [comment]
> I doubt that's going to happen in the next 2-3 weeks

I agree on that.  ... and I like DocFX a lot too _(such a Microsoft fan here)_ 😝 

> So we may have to make due with something else for 1.0 and then do something better for 2.0.

When I said _"until product becomes more mature"_ I meant **RESTier** to become more mature.  

So basically we both thinking the same way, use the easiest solution available for 1.0 and DocFX for 2.0.
  
However after your reply I feel like you found a way to host the solution and convert existing files to DocFx for 1.0.  It is great if that is the case here! 👍

**Comment by robertmclaws** [comment]
OH I gotcha. 👍 I'm pretty sure my ReadTheDocs documentation was Markdown already, so moving it to DocFX won't be too hard. The time-consuming part will be getting the rest of the Restier docs into Markdown and making it more understandable. Looking forward to what @superyyrrzz brings to the table too 👋.

**Comment by superyyrrzz** [comment]
> Those plans are still in their very early stages though, and haven't been cleared with MSFT yet.

@cilerler Sorry for not having knowledge about this, but I am willing to help if you have any question about DocFX 😄

**Comment by robertmclaws** [comment]
Hey everyone, Christmas has come a few days early this year! I've been hard at work cleaning up the codebase and getting ready for the next Beta release. All the 'Softies are home for the holidays, so don't expect an _official_ release until sometime in January. But I was able to get a NuGet feed for nightly builds going through an Azure Pipelines CI process. You can get the first nightly release from the feed at https://www.myget.org/F/restier-nightly/api/v3/index.json

There are breaking changes in here, and I haven't had the chance to document them all yet, but you're welcome to pull the code in and test against your services. [The 1.0 release project](https://github.com/OData/RESTier/projects/1) has more details, but the biggest thing to be aware of is that [the method pluralization has now been reversed](https://github.com/OData/RESTier/issues/571), so check your interceptors.

Pull it in, update your EntityFramework and OData packages to the very latest versions (YAY!), and update any of your current issues with your new findings.

Thanks, and Merry Christmas / Happy Holidays!

**Comment by cilerler** [comment]
@robertmclaws THANK YOU VERY MUCH! 🤗 I'm going to play with it tonight and will provide you a feedback.

By the way, would you please remove the **Welcome to GitHub Projects** card from the [Project 1](https://github.com/OData/RESTier/projects/1).  Thanks in advance...

**Comment by robertmclaws** [comment]
@cilerler So I didn't take it out (because it has some tips that I want to remember while I get used to that tool), but I did move it to "Done" so it's not in the way. Enjoy!

**Comment by robertmclaws** [comment]
By the way, since Restier does now have a future, I'm going to close this issue. But I'm not going to lock it or anything, so it can still be used for discussion about the future. :)

**Comment by mikepizzo** [comment]
Thanks for all of your work on this @robertmclaws!

**Comment by vitaly-livshits** [comment]
Could someone from the RESTier team please provide an update on the 1.0 release? Is there a target date set for it? We have clients on beta who would love to upgrade ASAP. Thanks!

**Comment by robertmclaws** [comment]
Here's where we're at:

- We're currently transitioning the project from Microsoft to the .NET Foundation. Microsoft is managing that process, so it's 100% out of my hands. Hoping to get an update from them soon.
- In the meantime, we've been working on cleaning up the code, fixing numerous bugs, and trying to get things polished up for release.
- We don't want to ship without better documentation. I've never been happy with the OData docs experience, and there are too many dark corners of RESTier that don't even have code comments, let alone understandable docs. That has to change.
- We're dependent on several layers of the OData stack, and believe it's best to ship after the next release of ODL (early March) and WebAPI (hopefully early April).

Given all the moving parts, I think we can ship a 1.0 release by mid-April. Maybe sooner if Microsoft is able to get a few things done quickly.

Having said that, we have a new build process, am simplified project structure, and a new Nightly build feed that is fairly stable. Please feel free to use the NuGet feed from the home page to test out the current builds and open a new issue if you find a problem. I'm very interested in any blockers you encounter.

Hit me up if you have any questions, happy to help. I hope that information is useful.

**Comment by grootstebozewolf** [comment]
@robertmclaws Main question is, how can we speed up this process and is the current beta stable enough for production? It's been a month since there been any updates on this repro.

**Comment by robertmclaws** [comment]
@grootstebozewolf I do not have the authority (or the code signing keys) to publish an official release until the .NET Foundation transfer is complete. So as I mentioned in my last post, it is 100% out of my hands, meaning there is nothing you can do to speed that up. MSLegal has the ball and is running with it.

In the meantime, don't let the lack of checkins fool you... it just means I have nothing that is ready for public consumption yet. However, I did recently complete a massive refactor that stabilized a bunch of the platform and reduced memory consumption, which has been published to the new CI build feed. I can't call them official releases, but they are being used in two Production deployments for the State of Florida this coming Tuesday, so take that for what it's worth.

Please install the new build in your app and test it out. The only major issue currently is that the OData WebAPI 7.X branch has a bug that prevents Functions (HTTP GET methods) with parameters from working. I submitted a fix, and that will go out with the next WebAPI release (which I've been told will be sometime in March). The workaround is to convert your GETs to POSTs and register the method as an Action instead.

Please let me know if you have any questions. Thanks!

**Comment by fileman** [comment]
@robertmclaws is there a roadmap for asp.net core 3.0 ?

**Comment by cilerler** [comment]
@fileman in last stand-up, he stated that RESTier would most likely support CORE3 in 2019Q3 _(right after that the CORE3 release)_

**Comment by fileman** [comment]
Support for RESTier users was a mirage... I'm interested at the code to try it with EF Core CosmosDB provider

**Comment by robertmclaws** [comment]
I'm sorry you feel that way @fileman. We've tried to be clear since taking over that .NET Core support will come _after_ we get the current codebase out the door. You have to remember, this project started 5 years ago, and STILL has not seen its first supported release. This codebase is so old, the best thing we can do to honor the work done by those before us is to get what we have out the door so we can start clean and quickly rebuild it as a .NET Core-native framework.

I'm working on documentation to show how you can have the API be .NET Framework and your app be .NET Core without compomising your architecture. I'll have more to share soon.

**Comment by fileman** [comment]
> I'm sorry you feel that way @fileman. 
>
@robertmclaws I'll try to explain better the intent of my question 
>We've tried to be clear since taking over that .NET Core support will come _after_ we get the current codebase out the door.
>
I know it, I think you have started the migration in some private repo, while it's moving to the .net foundation 
>You have to remember, this project started 5 years ago, and STILL has not seen its first supported release.
>
I remember well and I'm referring to this when I say it's a mirage :-) 
> This his codebase is so old, the best thing we can do to honor the work done by those before us is to get what we have out the door so we can start clean and quickly rebuild it as a .NET Core-native framework.
> 
We have honored and [extend](https://www.nuget.org/packages/AdvancedREI.Restier.Extensions/1.0.0-beta2a) their work, now it's time to honor your work and contribute to the success of project
> I'm working on documentation to show how you can have the API be .NET Framework and your app be .NET Core without compomising your architecture. I'll have more to share soon.
>
This is very interesting... soon = few days? ;-)

**Comment by robertmclaws** [comment]
Hey everyone! 1124 days and 117 checkins later, we've FINALLY released [Restier RC1](https://github.com/OData/RESTier/releases/tag/1.0.0-rc1.20191003.1)! We're still working through some issues to get it published to NuGet.org, but it's available now in the Nightly feed.

The RC1 announcement has an extensive list of changes to be aware of while you upgrade your apps, but PLEASE download the latest bits and try them out. If you have any problems, please open a new issue and label it "rc1".

The only thing left between here and RTM is getting the NuGet issues worked out (being able to publish Microsoft.* packages from our feed, deprecating old packages, etc), fixing any showstoppers, and documentation. I would like to make all that happen by the end of October, but it will depend on your feedback.

We hope you folks like the changes. ONWARD TO RTM!

**Comment by robertmclaws** [comment]
@fileman, thanks for your comments. I just wanted to point out that that I'm the one that shipped the AdvancedREI.Restier.* NuGet packages, so I am, in fact, honoring my work and contributing to the success of the project ;).

Please try out the new bits and let us know if you have any problems.
